### ISSUE TITLE
Hyper-parameter training optimization

### ISSUE THREAD

**Issue by alec-roberson** [alec-roberson/MS-CV2#6]
There are a lot of parameters when training a neural network (learning rate, weight decay, momentum...) and finding the best way to optimize these can have massive benefits.

The project here is to find and implement a scheme to optimize the hyper parameters. Talk to Daniel if you need help getting started.
